src/AppInstallerCommonCore/Public/winget/ThreadGlobals.h

```diff
@@ -0,0 +1,52 @@
+#pragma once
+
+#include <AppInstallerLogging.h>
+#include <AppInstallerTelemetry.h>
+#include <mutex>
+
+namespace AppInstaller::ThreadLocalStorage
+{
+
+    struct PreviousThreadGlobals;
+
+    struct ThreadGlobals
+    {
+        ThreadGlobals() = default;
+        ~ThreadGlobals() = default;
+
+        AppInstaller::Logging::DiagnosticLogger& GetDiagnosticLogger();
+
+        AppInstaller::Logging::TelemetryTraceLogger& GetTelemetryLogger();
+
+        // Set Globals for Current Thread
+        // Return RAII object with it's ownership to set the AppInstaller ThreadLocalStorage back to previous state
+        AppInstaller::ThreadLocalStorage::PreviousThreadGlobals* SetForCurrentThread();
```

Comment:
Here is why `auto` wouldn't work for my previous comment; this should be returning the RAII object itself (not a type that does RAII but itself has no lifetime semantic):
```suggestion
        std::unique_ptr<AppInstaller::ThreadLocalStorage::PreviousThreadGlobals SetForCurrentThread();
```